### ISSUE TITLE
Update components which interface with old React context API

### ISSUE THREAD

**Issue by bjankord** [cerner/terra-core#2477]
# Feature Request

## Description
In a lot of the terra components that interface with react-intl, [we've been interfacing with the old React context API](https://github.com/cerner/terra-core/blob/master/packages/terra-demographics-banner/src/DemographicsBanner.jsx#L67-L74) directly. We should update the components to use an abstraction like `FormattedMessage` or `injectIntl` to interface with react-intl's Intl context. This should allows us to more easily update to react-intl v3 when it is released.

More info on how we recommend to build components with `FormattedMessage` and/or `injectIntl`: https://engineering.cerner.com/terra-ui/#/getting-started/terra-ui/internationalization/building-components-which-include-translations

**Comment by bjankord** [comment]
We've reverted the PR as we had seen it introduced some unintended breaking changes. We are planning to talk through some updates to see if we can make this update passive.

**Comment by bjankord** [comment]
We talked in the terra leadership sync meeting this week and want to move forward with using the render prop version of `<FormattedTranslations />`

```jsx
import { FormattedMessage } from 'react-intl';
import Button from 'terra-button';

const Toggle = () => (
  <FormattedMessage id="App.toggle">
    {txt => (
      <Button text={txt} />
    )}
  </FormattedMessage>
);
```

If we are already using `injectIntl` in a component, we can continue using that.

The thinking behind this is that we believe this should be a more passive update instead of wrapping components with `injectIntl`. If you have questions about this change, feel free to post a comment back on this issue.

**Comment by bjankord** [comment]
Resolved with #2560
